src/typescript/frontend/src/components/header/wallet-button/ConnectWalletButton.tsx

```diff
@@ -62,39 +66,53 @@ export const ButtonWithConnectWalletFallback: React.FC<ConnectWalletProps> = ({
 
   // This component is used to display the `Connect Wallet` button and text with a scramble effect.
   // We use it in both mobile and desktop components.
-  return (
-    <>
-      {!connected || !children ? (
-        <Button
-          className={
-            className + (mobile ? " px-[9px] border-dashed border-b border-b-dark-gray" : "")
-          }
-          onClick={(e) => {
-            e.preventDefault();
-            onClick ? onClick() : openWalletModal();
-            handleReplay();
-          }}
-          onMouseOver={handleReplay}
+  const inner =
+    !connected || !children || isBanned ? (
+      <Button
+        className={
+          className + (mobile ? " px-[9px] border-dashed border-b border-b-dark-gray" : "")
+        }
+        disabled={isBanned}
+        onClick={(e) => {
+          e.preventDefault();
+          onClick ? onClick() : openWalletModal();
```

Comment:
```suggestion
          (onClick ?? openWalletModal)();
```

I think this is my code :P definitely prefer the nullish coalescing operator for readability